File: Klanvision_V5_Backend-master/models/Candidate.js
import { DataTypes } from 'sequelize';
import sequelize from '../config/db.js';

const Candidate = sequelize.define('Candidate', {
  id: {
    type: DataTypes.BIGINT,
    primaryKey: true,
    autoIncrement: true
  },
  name: {
    type: DataTypes.STRING
  },
  email: {
    type: DataTypes.STRING,
    allowNull: false
  },
  password: {
    type: DataTypes.STRING,
    allowNull: false
  },
  phone: {
    type: DataTypes.STRING
  },
  dob: {
    type: DataTypes.STRING
  },
  gender: {
    type: DataTypes.STRING
  },
  qualification: {
    type: DataTypes.STRING
  },
  experience: {
    type: DataTypes.STRING
  },
  skills: {
    type: DataTypes.TEXT
  },
  linkedin: {
    type: DataTypes.STRING
  },
  portfolio: {
    type: DataTypes.STRING
  },
  resumeData: {
    type: DataTypes.BLOB('long'),
    field: 'resume_data'
  },
  resumeFileName: {
    type: DataTypes.STRING,
    field: 'resume_file_name'
  },
  resumeContentType: {
    type: DataTypes.STRING,
    field: 'resume_content_type'
  },
  createdAt: {
    type: DataTypes.DATE,
    field: 'created_at',
    defaultValue: DataTypes.NOW
  }
}, {
  tableName: 'candidates',
  timestamps: false
});

export default Candidate;
export { Candidate };

File: Klanvision_backend/src/main/java/com/klanvision/backend/model/Candidate.java
package com.klanvision.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "candidates")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Candidate {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @Column(unique = true, nullable = false)
    private String email;

    @Column(nullable = false)
    private String password;

    private String phone;
    private String dob;
    private String gender;
    private String qualification;
    private String experience;

    @Column(columnDefinition = "TEXT")
    private String skills;

    private String linkedin;
    private String portfolio;

    @Lob
    @Column(columnDefinition = "LONGBLOB")
    private byte[] resumeData;
    
    private String resumeFileName;
    private String resumeContentType;

    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}
